### ISSUE TITLE
Deleted files remaining around on subsequent backups forever, whenever there's any kind of IO error (eg $HOME/.cache/doc submount)

### ISSUE THREAD

**Issue by spacelama** [backuppc/backuppc#534]
TL;DR: Please add "--ignore-errors" to default rsync options, as well as clarifying why you would want to do such a thing in README and release-notes (people should backport this manually to their own non-default configs because having deleted files hang around post-recovery is a form of data-loss).

Versions:
version 4.4.0, linux server, linux client

Partial clone of #87, which was closed without me being able to re-open it with my possible explanation.  "Kirby" also replied to me offlist saying my explanation fit his circumstances in the thread here: https://sourceforge.net/p/backuppc/mailman/backuppc-users/thread/d917d7bc-2a2e-637f-d467-bf2f652fd511%40gmail.com/#msg58787185

Paraphrasing the original comment:

If there's a virtual file system (eg, .sshfs, /home/$user/.cache/doc), or file otherwise inaccessible to the backup user, and it's always been like that, it's not an error not to back that up. The failure will be present for every backup. But rsync with backuppc's default flags turns off the codepath to tell backuppc about all file deletions for that backup run. Thus there will be a forever growing list of files if you're only ever doing incrementals, and no files will ever be deleted. Sure, I can add an infinite list of exclusions to my config, but that doesn't fix the problem where someone deletes a file from the filesystem after the rsync scan has been performed but before the file has been transferred, and now suddenly all file deletions from that backup are silently ignored.

It suffices to add "--ignore-errors" to rsync's default args. It is never a problem to send file deletions even in the case of IO errors when your target is a backup host where previous backups are not destroyed by the act of sending a new backup. Sure, flag these errors in the logfile, but don't cause all file deletions to be silently ignored!

**Comment by GWHAYWOOD** [comment]
Closed by 7c7ba54.

**Comment by spacelama** [comment]
> Closed by [7c7ba54](https://github.com/backuppc/backuppc/commit/7c7ba54fe09473647f056c5ae84e9576f5750c74).

For posterity, ITYM [2810d97](https://github.com/backuppc/backuppc/commit/2810d97fda8f7565640c7e77698982eb1fea190c)

Thanks!

**Comment by GWHAYWOOD** [comment]
Ah, yes, thanks.  Long day.
